test/end-to-end/company.js

```diff
@@ -228,7 +228,571 @@ describe("Company endpoint", () => {
+                }),
+            ]);
+
+            test_offer_hidden = await Offer.create(
+                generateTestOffer({
+                    isHidden: true,
+                    owner: test_company_with_hidden_offer._id.toString(),
+                    ownerName: test_company_with_hidden_offer.name,
+                    ownerLogo: test_company_with_hidden_offer.logo,
+                })
+            );
+        });
+
+        afterEach(async () => {
+            await test_agent.delete("/auth/login").expect(HTTPStatus.OK);
+        });
+
+        describe("Without Auth", () => {
+            test("should fail if invalid id", async () => {
+                const res = await test_agent
+                    .get("/company/123")
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+
+                expect(res.body).toHaveProperty(
+                    "error_code",
+                    ErrorTypes.VALIDATION_ERROR
+                );
+                expect(res.body).toHaveProperty(
+                    "errors",
+                    expect.arrayContaining([
+                        expect.objectContaining({
+                            param: "companyId",
+                            msg: ValidationReasons.OBJECT_ID,
+                        }),
+                    ])
+                );
+            });
+
+            test("should fail if company does not exist", async () => {
+                const id = "111111111111111111111111";
+                const res = await test_agent
+                    .get(`/company/${id}`)
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+
+                expect(res.body).toHaveProperty(
+                    "error_code",
+                    ErrorTypes.VALIDATION_ERROR
+                );
+                expect(res.body).toHaveProperty(
+                    "errors",
+                    expect.arrayContaining([
+                        expect.objectContaining({
+                            param: "companyId",
+                            msg: ValidationReasons.COMPANY_NOT_FOUND(id),
+                        }),
+                    ])
+                );
+            });
+
+            test("should succeed when the company has no offers", async () => {
+                const res = await test_agent
+                    .get(`/company/${test_company_without_offers.id}`)
+                    .expect(HTTPStatus.OK);
+
+                expect(res.body).toHaveProperty("offers", []);
+                expect(res.body).toHaveProperty(
+                    "company._id",
+                    test_company_without_offers._id.toString()
+                );
+            });
+
+            test("should return all offers when below limit", async () => {
+                const offers = await createTestOffers(
+                    CompanyConstants.offers.max_profile_visible - 1,
+                    test_company_with_offers_below_limit
+                );
```

Comment:
I do not feel too confortable with this style of creating offers but it is not technically wrong, in my opinion.